src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs

```diff
@@ -84,21 +84,28 @@ public static void EmitRequestHandler(this Endpoint endpoint, CodeWriter codeWri
         {
             return;
         }
-        if (!endpoint.Response.HasNoResponse)
+        if (endpoint.Response.IsAwaitable)
         {
-            codeWriter.Write("var result = ");
+            codeWriter.WriteLine($"var task = handler({endpoint.EmitArgumentList()});");
         }
-        if (endpoint.Response.IsAwaitable)
+        if (endpoint.Response.IsAwaitable && endpoint.Response.ResponseType?.NullableAnnotation == NullableAnnotation.Annotated)
+        {
+            codeWriter.WriteLine("if (task == null)");
```

Comment:
This feels odd to explicitly protect against, but you've already done the work 🤷‍♂️

Reply:
RDF has similar checks you can see [here](https://github.com/dotnet/aspnetcore/blob/main/src/Http/Http.Extensions/src/RequestDelegateFactory.cs#L2277).

Admittedly, this is an extremely defensive check against people writing endpoints like:

```
app.MapGet("/", (int id) => id == 0 ? SomeThingThatReturnsAValueTaskOfT() : SomethingThatReturnsNull);
```